### PR TITLE
Remove Deprecated sdsUdsPath from MeshConfig

### PR DESCRIPTION
**Pull request by nschhina**
Ref: https://github.com/istio/istio/issues/24410

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/config/constants/constants.go [hunk 1]
```diff
@@ -89,4 +89,8 @@ const (
 	// PodInfoAnnotationsPath is the filepath that pod annotations will be stored
 	// This is typically set by the downward API
 	PodInfoAnnotationsPath = "./etc/istio/pod/annotations"
+
+	// DefaultSdsUdsPath is the path used for SDS communication between istio-agent and istiod during
```
Comment (ramaraochavali) [inline pkg/config/constants/constants.go:93]: nit: This comment seems incorrect - between istio-agent and proxy?

File: pkg/kube/inject/inject.go [hunk 2]
```diff
@@ -780,7 +782,7 @@ func IntoObject(sidecarTemplate string, valuesConfig string, revision string, me
 	// due to bug https://github.com/kubernetes/kubernetes/issues/57923,
 	// k8s sa jwt token volume mount file is only accessible to root user, not istio-proxy(the user that istio proxy runs as).
 	// workaround by https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
-	if meshconfig.SdsUdsPath != "" {
+	if constants.DefaultSdsUdsPath != "" {
```
Comment (ramaraochavali) [inline pkg/kube/inject/inject.go:785]: Do we need this check?

File: pkg/kube/inject/webhook.go [hunk 3]
```diff
@@ -699,7 +701,7 @@ func (wh *Webhook) inject(ar *kubeApiAdmission.AdmissionReview, path string) *ku
 	// due to bug https://github.com/kubernetes/kubernetes/issues/57923,
 	// k8s sa jwt token volume mount file is only accessible to root user, not istio-proxy(the user that istio proxy runs as).
 	// workaround by https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
-	if wh.meshConfig.SdsUdsPath != "" {
+	if constants.DefaultSdsUdsPath != "" {
```
Comment (ramaraochavali) [inline pkg/kube/inject/webhook.go:704]: same here?

File: pkg/kube/inject/inject.go (hunk 2 above)
Comment (nschhina) [reply pkg/kube/inject/inject.go:785]: We don't need this check. I tried to cleanup this codepath in a different PR but it was blocked by different bootstrap config for TestEnvoy which didn't have sds cluster

File: pkg/kube/inject/webhook.go (hunk 3 above)
Comment (nschhina) [reply pkg/kube/inject/webhook.go:704]: same as above

File: pkg/kube/inject/inject.go (hunk 2 above)
Comment (nschhina) [reply pkg/kube/inject/inject.go:785]: Oh sorry, this is a different location- yeah we don't need this check I will remove it

**Pull request merged**